### PR TITLE
Add m2-recipes

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/m2-recipes/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
+package:
+  name: m2-binary-packages
+  version: 20230914
```
Comment (h-vetinari) [inline recipes/m2-recipes/meta.yaml:3]: Since this epoch appears all over, it might be a good idea to set a variable and reuse that?

Comment (isuruf) [reply recipes/m2-recipes/meta.yaml:3]: This file is auto-generated by the script given. I don't intend to make changes to it.

**Comment by isuruf** [comment]
@conda-forge/core, this is ready for review. I have initially added 3 outputs for easy review. I'll turn it into the full 149 outputs after merge in the feedstock.

**Review by beckermr — APPROVED** [review]

**Comment by carterbox** [comment]
When I download the Windows artifacts from Azure, there are no conda packages?

**Comment by isuruf** [comment]
@carterbox, all of them are `noarch` packages because we want them to be installable on win-64 and win-arm64 and the artifact upload here does not consider uploading noarch packages from windows.

**Review by xhochy — APPROVED** [review]

**Review by beckermr — CHANGES_REQUESTED** [review]
We're gonna try to make a special feedstock for these.

**Comment by jakirkham** [comment]
Is the plan now to add these here ( https://github.com/conda-forge/m2-builds )?

**Pull request merged**